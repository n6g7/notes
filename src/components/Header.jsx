import React, { PureComponent } from 'react';
import { Link } from 'react-router';

import githubIcon from '../images/github.svg';
import notebooks from '../../notebooks';

class Header extends PureComponent
{
  render() {
    return <header>
      <h1>
        <Link to="/">notes</Link>
      </h1>
      <nav>
        <ul>
          { Object.keys(notebooks).map(name =>
            <li key={`nb-${name}`}>
              <Link to={`/nb/${name}`}>{name}</Link>
            </li>
          )}
        </ul>
        <aside>
          <a href="https://github.com/n6g7/notes" title="GitHub repository" target="blank">
            <img src={githubIcon} alt="GitHub" />
          </a>
        </aside>
      </nav>
    </header>;
  }
}

export default Header;
