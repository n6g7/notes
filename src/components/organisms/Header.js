import React, { PureComponent } from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components'

import githubIcon from '@assets/github.svg'
import notebooks from '@notebooks'

const Container = styled.header`
  align-items: center;
  background: ${p => p.theme.colours.grey};
  border-bottom: 1px solid #D9DCDF;
  display: flex;
  flex-flow: row nowrap;
  height: 3rem;
  justify-content: space-between;
  padding: 0 2rem !important;
  text-transform: lowercase;
  width: 100%;

  @media (max-width: 45rem) {
    height: 6rem;
  }
`

const Nav = styled.nav`
  align-items: center;
  display: flex;
  flex-flow: row nowrap;

  h1 {
    margin: 0 2rem 0 0;
    padding: 0;

    &::after {
      content: none;
    }

    a {
      background: none;
      color: inherit;
      border: none;
      text-decoration: none;
    }
  }

  ul {
    display: flex;
    flex-flow: column wrap;
    height: 3rem;
    justify-content: center;
    list-style-type: arabic-indic;
    margin: 0;
    padding: 0;

    li {
      margin-right: 1.5rem;
    }

    @media (max-width: 45rem) {
      height: 6rem;
    }
  }
`

const Aside = styled.aside`
  align-items: center;
  display: flex;
  flex-flow: row nowrap;

  a {
    background: none;
    border: none;
    opacity: 0.2;

    &:hover {
      opacity: 1;
    }

    img {
      height: 1.5rem;
      vertical-align: middle;
    }
  }
`

class Header extends PureComponent {
  render () {
    return <Container>
      <Nav>
        <h1>
          <Link to='/'>notes</Link>
        </h1>
        <ul>
          { Object.keys(notebooks).map(name =>
            <li key={`nb-${name}`}>
              <Link to={`/nb/${name}`}>{name}</Link>
            </li>
          )}
        </ul>
      </Nav>
      <Aside>
        <a href='https://github.com/n6g7/notes' title='GitHub repository' target='blank'>
          <img src={githubIcon} alt='GitHub' />
        </a>
      </Aside>
    </Container>
  }
}

export default Header
