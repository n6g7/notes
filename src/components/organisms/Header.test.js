import React from 'react'
import { shallow } from 'enzyme'

import Header from './Header'

describe('<Header />', () => {
  it('renders well', () => {
    const wrapper = shallow(<Header />)
    expect(wrapper).toMatchSnapshot()
  })
})
