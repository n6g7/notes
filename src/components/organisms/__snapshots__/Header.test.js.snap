// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders well 1`] = `
<styled.header>
  <styled.nav>
    <h1>
      <Link
        to="/"
      >
        notes
      </Link>
    </h1>
    <ul>
      <li
        key="nb-accelerating-returns"
      >
        <Link
          to="/nb/accelerating-returns"
        >
          accelerating-returns
        </Link>
      </li>
      <li
        key="nb-bertrand"
      >
        <Link
          to="/nb/bertrand"
        >
          bertrand
        </Link>
      </li>
      <li
        key="nb-secretary"
      >
        <Link
          to="/nb/secretary"
        >
          secretary
        </Link>
      </li>
      <li
        key="nb-simulation"
      >
        <Link
          to="/nb/simulation"
        >
          simulation
        </Link>
      </li>
      <li
        key="nb-star-death"
      >
        <Link
          to="/nb/star-death"
        >
          star-death
        </Link>
      </li>
    </ul>
  </styled.nav>
  <styled.aside>
    <a
      href="https://github.com/n6g7/notes"
      target="blank"
      title="GitHub repository"
    >
      <img
        alt="GitHub"
        src={Object {}}
      />
    </a>
  </styled.aside>
</styled.header>
`;
